test/terraform_azure_aks_example_test.go

```diff
@@ -0,0 +1,104 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTerraformAzureAKSExample(t *testing.T) {
+	t.Parallel()
+	// MC_+ResourceGroupName_ClusterName_AzureRegion must be no greater than 80 chars.
+	// https://docs.microsoft.com/en-us/azure/aks/troubleshooting#what-naming-restrictions-are-enforced-for-aks-resources-and-parameters
+	expectedClusterName := fmt.Sprintf("terratest-aks-cluster-%s", random.UniqueId())
+	expectedResourceGroupName := fmt.Sprintf("terratest-aks-rg-%s", random.UniqueId())
+	expectedAagentCount := 3
+
+	terraformOptions := &terraform.Options{
+		TerraformDir: "../examples/terraform-azure-aks-example",
+		Vars: map[string]interface{}{
+			"cluster_name":        expectedClusterName,
+			"resource_group_name": expectedResourceGroupName,
+			"agent_count":         expectedAagentCount,
+		},
+	}
+	// At the end of the test, run `terraform destroy` to clean up any resources that were created
+	defer terraform.Destroy(t, terraformOptions)
+
+	// This will run `terraform init` and `terraform apply` and fail the test if there are any errors
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Look up the cluster node count
+	var ActualCount int32 = 0
+	cluster, err := azure.GetManagedCluster(t, expectedResourceGroupName, expectedClusterName, "")
+	assert.Nil(t, err)
+	ActualCount = *(*cluster.ManagedClusterProperties.AgentPoolProfiles)[0].Count
```

Comment:
We can probably simplify this as:
 
```suggestion
	actualCount := *(*cluster.ManagedClusterProperties.AgentPoolProfiles)[0].Count
```
and remove the `ActualCount` declaration above.

Reply:
Done